### PR TITLE
Fix utf-8 conflict in expand_path_spec.rb

### PR DESCRIPTION
**Pull request by mjago**

### CODE REVIEW

File: core/file/expand_path_spec.rb [hunk 1]
```diff
@@ -167,14 +168,14 @@
...
-      weird_path = "\xde\xad\xbe\xaf".force_encoding Encoding::ASCII_8BIT
+      weird_path = [222, 173, 190, 175].pack('C*')
       File.expand_path(weird_path).encoding.should equal(Encoding::ASCII_8BIT)
     end
 
     platform_is_not :windows do
       it "expands a path when the default external encoding is ASCII-8BIT" do
         Encoding.default_external = Encoding::ASCII_8BIT
-        File.expand_path("\xde\xad\xbe\xaf", @rootdir).should == "#{@rootdir}\xde\xad\xbe\xaf"
+        File.expand_path( [222, 173, 190, 175].pack('C*'), @rootdir).should == "#{@rootdir}" +  [222, 173, 190, 175].pack('C*')
```
Comment (eregon) [inline core/file/expand_path_spec.rb]: Could you extract the common part here?

**Comment by eregon** [comment]
Thank you!

**Comment by backus** [comment]
@mjago I'm not sure what your goal is for these PRs, but if you're interested in eliminating more of these utf8 conflicts then here is a list of files that produce utf8 issues: https://github.com/mbj/mutant/blob/master/spec/integrations.yml#L10-L76

**Comment by mjago** [comment]
@backus Thanks for the heads-up - mutant looks an interesting project 👍

**Pull request merged**